website/main/content/blog/2025-07-14-scheduled-plot-lifecycle.md

```diff
@@ -0,0 +1,221 @@
+    `history_class_1 = H_1 % HISTORY_MODULO`. This determines a `plot_id_1` and assigns them to a
+    corresponding shard. They begin filling their drive with sectors committed to `H_1`.
+
+2.  **Maintenance:** As time passes and the chain grows to a history size of `H_2`, their initial
+    sectors from `H_1` begin to expire. The farmer uses the newly freed space to plot new sectors.
+    They now have a choice:
+
+- **Re-use Existing Plot:** The farmer can choose to remain in the same plot (and thus the same
+  shard). To do this, they find a `committed_history_size` (`H_commit`) that is less than or equal
+  to `H_2` and also satisfies `H_commit % HISTORY_MODULO == history_class_1`. A farmer can easily
+  calculate the most recent history size that meets this criterion.
+- **Utilize a New Plot:** Alternatively, the farmer can commit to the latest history, `H_2`. This
+  will generate a new class, `history_class_2 = H_2 % HISTORY_MODULO`, creating a new `plot_id` and
+  likely assigning them to a new shard.
+
+This model removes the concept of a mandatory, network-wide "Epoch Turnover." Instead, re-shuffling
+becomes a strategic choice made by the farmer.
+
+### Benefits and Parameter Tuning
+
+This deterministic mapping provides significant advantages:
+
+- **Simplicity:** The logic is extremely straightforward, relying on a single new parameter
+  (`HISTORY_MODULO`) and simple modulo arithmetic.
+- **Farmer Quality of Life:** Farmers are never forced into a disruptive, network-wide re-plotting
+  event. They have fine-grained control over which plots they maintain, reducing operational
+  overhead.
+- **Flexibility:** Farmers can choose to commit to a slightly older history size to preserve their
+  shard assignment or switch to a new one if it is more advantageous.
+
+The primary consideration is tuning the `HISTORY_MODULO` parameter, which involves a direct
+trade-off between farmer flexibility and preventing strategic shard selection.
+
+### A model for parameter tuning
+
+The key to this design is selecting an optimal value for $M$, the `HISTORY_MODULO`. This choice
+involves balancing two competing factors: **History Lag** and **Shard Selection Power**.
+
+Let's define the key parameters:
+
+- \\(M\\): The `HISTORY_MODULO`, the number of distinct history classes. **This is the value we want
+  to determine.**
+- \\(S\\): The total number of shards in the network.
+- \\(H\_{current}\\): The current `history_size` of the blockchain.
+
+#### Modeling Farmer Flexibility (History Lag)
+
+When a farmer wants to re-use an existing plot, they may not be able to use \\(H\_{current}\\). This
+creates a "lag" between their plotted history and the most current state of the network. We can
+quantify this as the **Average History Lag** (\(L_H\)):
+
+$$
+L_H = \frac{M}{2}
+$$
+
+A larger $M$ gives the farmer more classes to choose from, reducing the average lag for any given
```

Comment:
IIRC `$something$` is not rendered properly here yet

Reply:
Also if `M` (or `HISTORY_MODULO`, not sure why not use it directly) is larger, then the lag will be larger too. Or am I missing something?

Reply:
You are right, good catch, my muscle memory messed up with me there.

The instant lag for a single sector would increase, but assuming sectors being created at random instants of time, the average lag would be less because you have more opportunities to map your plotID closer to the latest history .